src/kubernetes.cc

```diff
@@ -1010,6 +1170,72 @@ void KubernetesReader::WatchNodes(
   LOG(INFO) << "Watch thread (node) exiting";
 }
 
+void KubernetesReader::ServiceCallback(
+    MetadataUpdater::UpdateCallback callback,
+    const json::Object* service, Timestamp collected_at, bool is_deleted)
+    throw(json::Exception) {
+  UpdateServiceToMetadataCache(service, is_deleted);
+
+  // TODO: using a temporary did not work here.
+  std::vector<MetadataUpdater::ResourceMetadata> result_vector;
+  result_vector.emplace_back(GetClusterMetadata(collected_at, is_deleted));
```

Comment:
Careful — this `is_deleted` refers to the service, right? You don't want to notify the API that the whole cluster has been deleted just because one service was. I would suggest just not having an `is_deleted` parameter in `GetClusterMetadata`.

Reply:
updated